src/binary-reader.cc

```diff
@@ -1605,6 +1605,9 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
       uint32_t subsection_size;
       in_u32_leb128(ctx, &name_type, "name type");
       in_u32_leb128(ctx, &subsection_size, "subsection size");
+      size_t subsection_end = ctx->offset + subsection_size;
+      if (subsection_end > ctx->read_end)
+        RAISE_ERROR("invalid sub-section size: extends past end");
```

Comment:
https://github.com/WebAssembly/design/pull/1016 seems to have good support and makes all strings utf8. If that's the case then perhaps calling it utf8-name is redundant.